### DISCUSSION TITLE
What would be the alternative to Jooq for Orientdb

### DISCUSSION THREAD

**Discussion by nithyasharabheshwara** [orientechnologies/orientdb#9708] [Q&A] [answered]
What would be the alternative to Jooq for Orientdb for constructing queries with a DSL. 
Gremlin?

**Comment by PhantomYdn** [comment, marked as answer]
Recommend to check this one: https://github.com/raymanrt/orientqb for query building.
Regarding DSL: you can try to reuse this one https://github.com/OrienteerBAP/wicket-orientdb/blob/master/wicket-orientdb/src/main/java/ru/ydn/wicket/wicketorientdb/utils/OSchemaHelper.java
Or there is Hibernate-like solution for OrientDB: https://github.com/OrienteerBAP/Transponder
There is no yet README, because it's in very active development, but you can check some samples here: https://github.com/OrienteerBAP/Transponder/tree/master/transponder-orientdb/src/test/java/org/orienteer/transponder/orientdb
